src/estimagic/optimization/tranquilo/region.py

```diff
@@ -16,31 +16,19 @@ class Region:
     radius: float
     bounds: Bounds = None
 
-    @property
-    def shape(self) -> str:
-        any_bounds_binding = _any_bounds_binding(
-            bounds=self.bounds, center=self.center, radius=self.radius
-        )
-        return "cube" if any_bounds_binding else "sphere"
-
-    @property
-    def cube_bounds(self) -> Bounds:
-        if self.shape == "sphere":
-            raise AttributeError(
-                "The trustregion is a sphere, and thus has no cube bounds."
-            )
-        radius = get_radius_of_cube_with_volume_of_sphere(self.radius, len(self.center))
-        bounds = _get_cube_bounds(center=self.center, radius=radius, bounds=self.bounds)
-        return bounds
-
-    @property
-    def cube_center(self) -> np.ndarray:
-        if self.shape == "sphere":
-            raise AttributeError(
-                "The trustregion is a sphere, and thus has no cube center."
-            )
-        center = _get_cube_center(bounds=self.cube_bounds)
-        return center
+    def __post_init__(self):
+        shape = _get_shape(self.center, self.radius, self.bounds)
+        cube_bounds = _get_cube_bounds(self.center, self.radius, self.bounds, shape)
+        cube_center = _get_cube_center(cube_bounds)
+        effective_center = _get_effective_center(shape, self.center, cube_center)
+        effective_radius = _get_effective_radius(shape, self.radius, cube_bounds)
+
+        # cannot use standard __setattr__ because it is frozen
+        super().__setattr__("shape", shape)
+        super().__setattr__("cube_bounds", cube_bounds)
+        super().__setattr__("cube_center", cube_center)
+        super().__setattr__("effective_center", effective_center)
+        super().__setattr__("effective_radius", effective_radius)
```

Comment:
I think `cube_bounds` and `cube_center` need to be private attributes that are accessed via corresponding properties. Currently we lost the error handling (i.e. `AttributeErrors if shape == "sphere"`)